### PR TITLE
Student Main Page Backend API

### PR DESCRIPTION
**Pull request by Mari-Hayashi**
Implemented the following APIs:

* api/job/get-recommended-jobs
* api/job/get-new-jobs

### CODE REVIEW

File: src/server/src/modules/job.ts [hunk 1]
```diff
@@ -481,3 +480,99 @@ export const getApplicants = async (
 
     return getApplicantsResult;
 };
+
+/**
+ * @description Returns at most 20 newest jobs that match with student's department
+ *              and class standing.
+ * @param {number} studentId - id of student
+ * @param {number} page - page index
```
Comment (johan1505) [inline src/server/src/modules/job.ts]: This function doesn't receive a page parameter

File: src/server/src/modules/job.ts [hunk 2]
```diff
@@ -481,3 +480,99 @@ export const getApplicants = async (
...
+ * @param {number} numOfItems - number of items per page
```
Comment (johan1505) [inline src/server/src/modules/job.ts]: This function doesn't receive a numOfItems parameter

File: src/server/src/modules/job.ts [hunk 3]
```diff
@@ -481,3 +480,99 @@ export const getApplicants = async (
...
+        .andWhere('job.id NOT IN (:...appliedJobIds)', {
+            // It causes a SQL parse error when an empty array is passed in.
+            appliedJobIds: appliedJobIds.length > 0 ? appliedJobIds : [-1],
+        })
+        .andWhere('(:departmentId < 0 OR job.departmentId = :departmentId)', {
+            departmentId: student.departmentId ?? -1
+        })
+        .andWhere(`(:classStanding = :null OR
+                    :classStanding = ANY(string_to_array(job.targetYears, :comma)))`, {
+            classStanding: student.classStanding ?? "NULL",
```
Comment (johan1505) [inline src/server/src/modules/job.ts:525]: what does "??" do? Never seen it before.

File: src/server/src/routes/job/index.ts [hunk 4]
```diff
@@ -390,6 +393,79 @@ router.get(
...
+
+interface GetNewJobsRequest extends Request {
+    query: {
+        page: string;
+        numOfItems: string;
+    };
+}
+
+router.get(
+    '/get-new-jobs',
```
Comment (johan1505) [inline src/server/src/routes/job/index.ts:409]: I thought we agreed the get new jobs api was not going to have pagination

**Review by johan1505 — CHANGES_REQUESTED** [review]

File: src/server/src/routes/job/index.ts (hunk 4 above)
Comment (Mari-Hayashi) [reply src/server/src/routes/job/index.ts:409]: No, we agreed that get recommended jobs API does not have pagination, but get new jobs API does.

File: src/server/src/modules/job.ts (hunk 3 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/job.ts:525]: student.classStanding ?? "NULL"

is same as

student.classStanding ? student.classStanding : "NULL"

File: src/server/src/modules/job.ts (hunk 1 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/job.ts]: Done.

File: src/server/src/modules/job.ts (hunk 2 above)
Comment (Mari-Hayashi) [reply src/server/src/modules/job.ts]: Done.

**Review by johan1505 — APPROVED** [review]

**Pull request merged**